source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/ColumnRenderer.java

```diff
@@ -156,30 +157,32 @@ else if ((column = Column.dynamic_cast(baseCardElement)) == null)
         }
         else
         {
-            if (TextUtils.isEmpty(columnSize) || columnSize.equals(g_columnSizeStretch))
-            {
-                layoutParams = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.MATCH_PARENT, 1);
-                returnedView.setLayoutParams(layoutParams);
-            }
-            else if (columnSize.equals(g_columnSizeAuto))
+            try
             {
-                layoutParams = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.WRAP_CONTENT, LinearLayout.LayoutParams.MATCH_PARENT);
+                // I'm not sure what's going on here
+                float columnWeight = Float.parseFloat(columnSize);
+                layoutParams = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.MATCH_PARENT);
+                layoutParams.width = 0;
+                layoutParams.weight = columnWeight;
                 returnedView.setLayoutParams(layoutParams);
             }
-            else
+            catch (NumberFormatException numFormatExcep)
             {
-                try
+                if (TextUtils.isEmpty(columnSize) || columnSize.equals(g_columnSizeStretch))
                 {
-                    // I'm not sure what's going on here
-                    float columnWeight = Float.parseFloat(columnSize);
-                    layoutParams = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.MATCH_PARENT);
-                    layoutParams.width = 0;
-                    layoutParams.weight = columnWeight;
+                    layoutParams = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.MATCH_PARENT, 1);
                     returnedView.setLayoutParams(layoutParams);
                 }
-                catch (NumberFormatException numFormatExcep)
+                else
                 {
-                    throw new IllegalArgumentException("Column Width (" + column.GetWidth() + ") is not a valid weight ('auto', 'stretch', <integer>).");
+                    // If the width is Auto or is not valid (not weight, pixel, empty or stretch)
+                    layoutParams = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.WRAP_CONTENT, LinearLayout.LayoutParams.MATCH_PARENT);
+                    returnedView.setLayoutParams(layoutParams);
+
+                    if (!columnSize.equals(g_columnSizeAuto))
+                    {
+                        renderedCard.addWarning(new AdaptiveWarning(AdaptiveWarning.INVALID_COLUMN_WIDTH_VALUE, "Column Width (" + column.GetWidth() + ") is not a valid weight ('auto', 'stretch', <integer>)."));
```

Comment:
Probably update the warning text to include what the behavior is defaulting to also for debuggability.